### PR TITLE
Good Practices

### PR DESCRIPTION
**Pull request by EvyatarD**

### CODE REVIEW

File: Good Coding Practices/src/css/styles.css [hunk 1]
```diff
@@ -19,10 +26,17 @@ body {
...
   border-radius: 50px;
-  background: blue;
+  /* animation-name: example;
+  animation-duration: 2s; */
+  transition-duration: 1s;
+
 }
+
+#piece:hover{
+background-color:white !important;
```
Comment (giladaya) [inline Good Coding Practices/src/css/styles.css]: Indentation and spacing!!!

File: Good Coding Practices/src/js/scripts.js [hunk 2]
```diff
@@ -1,44 +1,112 @@
...
     this.el = el;
   };
-  const moveDelta = function(dx, dy) {
+
+  const moveDelta = function (dx, dy) {
     const pos = this.el.getBoundingClientRect();
-    this.el.style.left = `${pos.left + dx}px`;
-    this.el.style.top = `${pos.top + dy}px`;
+    const position = [pos.top, pos.left];
+    if (dx === 0) {
```
Comment (giladaya) [inline Good Coding Practices/src/js/scripts.js]: This logic allows movement in one axis only for no good reason

File: Good Coding Practices/src/js/scripts.js [hunk 3]
```diff
@@ -1,44 +1,112 @@
...
+      this.el.style.top = `${movePos(position[0], dy,window.innerHeight - 100)}px`;
```
Comment (giladaya) [inline Good Coding Practices/src/js/scripts.js]: non uniform spacing

File: Good Coding Practices/src/js/scripts.js [hunk 4]
```diff
@@ -1,44 +1,112 @@
...
+    }
+    else
+      this.el.style.left = `${movePos(position[1], dx,window.innerWidth - 100)}px`;
```
Comment (giladaya) [inline Good Coding Practices/src/js/scripts.js]: Why 100? what does it mean? I don't like surprises

File: Good Coding Practices/src/js/scripts.js [hunk 5]
```diff
@@ -1,44 +1,112 @@
...
+      positionSide = screenSize;
+    else if (pos.top + px <= 0)
+      positionSide = 0;
+    return positionSide;
+  }
+
+  const pieceColor = function (temp) {
+    let color = "";
+    if (temp <= 10) {
+      color = "blue";
```
Comment (giladaya) [inline Good Coding Practices/src/js/scripts.js]: Hard coded style values

File: Good Coding Practices/src/js/scripts.js [hunk 6]
```diff
@@ -1,44 +1,112 @@
...
+    if ((pos + px < screenSize) && (pos + px > 0))
+      positionSide = pos + px;
+    else if (pos + px >= screenSize)
+      positionSide = screenSize;
+    else if (pos.top + px <= 0)
+      positionSide = 0;
+    return positionSide;
+  }
+
+  const pieceColor = function (temp) {
```
Comment (giladaya) [inline Good Coding Practices/src/js/scripts.js]: Function name is not very indicative of it's functionality

File: Good Coding Practices/src/js/scripts.js [hunk 7]
```diff
@@ -1,44 +1,112 @@
...
+        for (item in myJson) {
+          if (myJson[item].temp_c)
+            piece.pieceColor(Number(myJson[item].temp_c));
+        }
+      });
+  }
+
+  const resetLocation = function () {
+    this.el.style.left = "50%";
+    this.el.style.top = "100px";
```
Comment (giladaya) [inline Good Coding Practices/src/js/scripts.js]: Why 50%? Why 100px?

File: Good Coding Practices/src/js/scripts.js [hunk 8]
```diff
@@ -1,44 +1,112 @@
...
+    else if (temp > 20 && temp < 31)
+      color = "Yellow";
+    else if (temp > 30)
+      color = "Red";
+    this.el.style.background = color;
+    this.el.style.borderColor = color;
+  }
+
+  const weatherApi = function () {
+    fetch('http://api.apixu.com/v1/current.json?key=dda6e762ae4f41efb7e173552192204&q=tel%20aviv')
```
Comment (giladaya) [inline Good Coding Practices/src/js/scripts.js]: This knowledge does not belong in `piece`
(API URL, result data structure, API key!)

File: Good Coding Practices/src/js/scripts.js [hunk 9]
```diff
@@ -1,44 +1,112 @@
...
+      .then(function (response) {
+        return response.json();
+      })
+      .then(function (myJson) {
+        for (item in myJson) {
```
Comment (giladaya) [inline Good Coding Practices/src/js/scripts.js]: The data structure is fixed and well defined. No need for looping and searching.

File: Good Coding Practices/src/js/scripts.js [hunk 10]
```diff
@@ -1,44 +1,112 @@
...
 
 window.addEventListener("DOMContentLoaded", event => {
   piece.init(document.getElementById("piece"));
   init();
 });
+
+function init() {
+  piece.resetLocation();
+  piece.weatherApi();
+  setClickEvent(document.getElementById("btn-up"),0,-100);
```
Comment (giladaya) [inline Good Coding Practices/src/js/scripts.js]: Spacing!

File: Good Coding Practices/src/js/scripts.js [hunk 11]
```diff
@@ -1,44 +1,112 @@
...
+
+}
+
+
+function setClickEvent(elemBtn,x,y){
+  let elem = elemBtn;
+  elem.dataset.dx =x;
+  elem.dataset.dy = y;
+  elem.addEventListener("click", handleClick);
+  }
```
Comment (giladaya) [inline Good Coding Practices/src/js/scripts.js]: Indentation!

File: Good Coding Practices/src/js/scripts.js (hunk 7 above)
Comment (EvyatarD) [reply Good Coding Practices/src/js/scripts.js]: this initial location

File: Good Coding Practices/src/js/scripts.js (hunk 11 above)
Comment (EvyatarD) [reply Good Coding Practices/src/js/scripts.js]: I use ctrl+shift+F..

**Pull request closed without merging**